src/main/java/hudson/plugins/openid/OpenIdSsoSecurityRealm.java

```diff
@@ -160,12 +160,16 @@ public Authentication authenticate(Authentication authentication) throws Authent
     /**
      * The login process starts from here.
      */
-    public HttpResponse doCommenceLogin(@QueryParameter String from) throws IOException, OpenIDException {
+    public HttpResponse doCommenceLogin(@QueryParameter String from) throws IOException, OpenIDException, IllegalStateException {
```

Comment:
I looked up some examples of the `@Restricted` tag, and it appeared to be on methods used by the "do" methods rather than themselves.  For a few of these in openid, since all the work is done within the "do" method, I didn't know where to place the restriction.